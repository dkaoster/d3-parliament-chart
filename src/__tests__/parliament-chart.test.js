/* eslint-disable no-undef */
import parliamentChart from '../parliament-chart';

test('empty parliament chart', () => {
  expect(parliamentChart().data()).toMatchSnapshot();
});

test('basic parliament chart', () => {
  expect(parliamentChart([{}, {}, {}, {}], 800).data()).toMatchSnapshot();
});

test('aggregated parliament chart data', () => {
  expect(
    parliamentChart([], 800)
      .aggregatedData([{ seats: 12 }, { seats: 25 }, { seats: 5 }])
      .data(),
  ).toMatchSnapshot();
});

test('parliament chart data overflow', () => {
  expect(
    parliamentChart([], 400)
      .aggregatedData([{ seats: 60 }, { seats: 25 }, { seats: 5 }])
      .data(),
  ).toMatchSnapshot();
});

test('parliament chart data options', () => {
  expect(
    parliamentChart([], 800)
      .aggregatedData([{ seats: 12 }, { seats: 25 }, { seats: 5 }])
      .debug(true)
      .sections(5)
      .sectionGap(20)
      .seatRadius(20)
      .rowHeight(50)
      .data(),
  ).toMatchSnapshot();
});
