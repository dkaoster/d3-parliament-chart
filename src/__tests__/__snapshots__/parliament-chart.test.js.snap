// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aggregated parliament chart data 1`] = `
Array [
  Object {
    "x": 14.253411828439596,
    "y": 373.2439260226661,
  },
  Object {
    "x": 56.52130359641177,
    "y": 373.30605419378406,
  },
  Object {
    "x": 18.925275072339502,
    "y": 342.02018072571093,
  },
  Object {
    "x": 26.120225849628866,
    "y": 311.27963323788583,
  },
  Object {
    "x": 35.790626546417116,
    "y": 281.2258160609881,
  },
  Object {
    "x": 47.872449648296026,
    "y": 252.05771487023955,
  },
  Object {
    "x": 62.285701665131,
    "y": 223.96845103442433,
  },
  Object {
    "x": 78.93495276623452,
    "y": 197.14400296347145,
  },
  Object {
    "x": 127.64181276590767,
    "y": 201.605956393907,
  },
  Object {
    "x": 97.70996861921532,
    "y": 171.76197474941503,
  },
  Object {
    "x": 156.7619747494151,
    "y": 112.70996861921526,
  },
  Object {
    "x": 186.60595639390704,
    "y": 142.64181276590767,
  },
  Object {
    "x": 182.14400296347148,
    "y": 93.93495276623446,
  },
  Object {
    "x": 208.96845103442442,
    "y": 77.285701665131,
  },
  Object {
    "x": 237.0577148702396,
    "y": 62.87244964829597,
  },
  Object {
    "x": 267.59153240167893,
    "y": 95.3376817510948,
  },
  Object {
    "x": 266.2258160609881,
    "y": 50.790626546417116,
  },
  Object {
    "x": 296.27963323788595,
    "y": 41.12022584962881,
  },
  Object {
    "x": 327.02018072571093,
    "y": 33.9252750723395,
  },
  Object {
    "x": 358.3060541937841,
    "y": 71.52130359641177,
  },
  Object {
    "x": 358.2439260226662,
    "y": 29.253411828439596,
  },
  Object {
    "x": 441.75607397733387,
    "y": 29.253411828439596,
  },
  Object {
    "x": 441.69394580621594,
    "y": 71.52130359641177,
  },
  Object {
    "x": 472.9798192742892,
    "y": 33.9252750723395,
  },
  Object {
    "x": 503.7203667621141,
    "y": 41.120225849628866,
  },
  Object {
    "x": 533.7741839390119,
    "y": 50.790626546417116,
  },
  Object {
    "x": 532.4084675983211,
    "y": 95.3376817510948,
  },
  Object {
    "x": 562.9422851297604,
    "y": 62.87244964829597,
  },
  Object {
    "x": 591.0315489655757,
    "y": 77.285701665131,
  },
  Object {
    "x": 617.8559970365285,
    "y": 93.93495276623446,
  },
  Object {
    "x": 613.394043606093,
    "y": 142.64181276590767,
  },
  Object {
    "x": 643.2380252505849,
    "y": 112.70996861921532,
  },
  Object {
    "x": 702.2900313807847,
    "y": 171.7619747494151,
  },
  Object {
    "x": 672.3581872340924,
    "y": 201.60595639390704,
  },
  Object {
    "x": 721.0650472337655,
    "y": 197.14400296347154,
  },
  Object {
    "x": 737.714298334869,
    "y": 223.96845103442433,
  },
  Object {
    "x": 752.127550351704,
    "y": 252.0577148702396,
  },
  Object {
    "x": 764.2093734535829,
    "y": 281.2258160609881,
  },
  Object {
    "x": 773.8797741503712,
    "y": 311.27963323788595,
  },
  Object {
    "x": 781.0747249276606,
    "y": 342.0201807257109,
  },
  Object {
    "x": 743.4786964035882,
    "y": 373.3060541937841,
  },
  Object {
    "x": 785.7465881715605,
    "y": 373.24392602266613,
  },
]
`;

exports[`basic parliament chart 1`] = `
Array [
  Object {
    "x": 41.53474138562075,
    "y": 266.5188282423451,
  },
  Object {
    "x": 251.5188282423452,
    "y": 56.53474138562075,
  },
  Object {
    "x": 548.4811717576549,
    "y": 56.53474138562075,
  },
  Object {
    "x": 758.4652586143793,
    "y": 266.5188282423452,
  },
]
`;

exports[`empty parliament chart 1`] = `Array []`;

exports[`parliament chart data options 1`] = `
Array [
  Object {
    "x": 21.178703511681988,
    "y": 375.0930555403959,
  },
  Object {
    "x": 71.35524186565772,
    "y": 375.1232038059689,
  },
  Object {
    "x": 30.506650767759652,
    "y": 316.2606914995297,
  },
  Object {
    "x": 86.15134962259935,
    "y": 303.02439185626736,
  },
  Object {
    "x": 133.70417543735704,
    "y": 318.4752415750147,
  },
  Object {
    "x": 48.91394432739156,
    "y": 259.60886714709835,
  },
  Object {
    "x": 116.55966536847353,
    "y": 235.99829378385468,
  },
  Object {
    "x": 75.94827241531476,
    "y": 206.52965498749955,
  },
  Object {
    "x": 111.10599420428548,
    "y": 158.13920235220485,
  },
  Object {
    "x": 151.68194574567224,
    "y": 187.65662206695893,
  },
  Object {
    "x": 153.23325799780866,
    "y": 116.02564985517495,
  },
  Object {
    "x": 206.0308667434839,
    "y": 138.02439185626736,
  },
  Object {
    "x": 235.42012935810754,
    "y": 178.4752415750147,
  },
  Object {
    "x": 201.42417817782214,
    "y": 81.01289688068346,
  },
  Object {
    "x": 270.0286629178885,
    "y": 101.67269899152149,
  },
  Object {
    "x": 254.49458714317547,
    "y": 53.96129155125209,
  },
  Object {
    "x": 311.38097596474285,
    "y": 35.47778337556139,
  },
  Object {
    "x": 326.85770633059957,
    "y": 83.20782346865673,
  },
  Object {
    "x": 342.33696665875766,
    "y": 131.00187120002664,
  },
  Object {
    "x": 370.2163733796186,
    "y": 26.168988089230368,
  },
  Object {
    "x": 400,
    "y": 75,
  },
  Object {
    "x": 429.7836266203815,
    "y": 26.168988089230368,
  },
  Object {
    "x": 457.66303334124234,
    "y": 131.00187120002664,
  },
  Object {
    "x": 473.1422936694005,
    "y": 83.20782346865673,
  },
  Object {
    "x": 488.6190240352572,
    "y": 35.477783375561444,
  },
  Object {
    "x": 545.5054128568246,
    "y": 53.96129155125209,
  },
  Object {
    "x": 529.9713370821116,
    "y": 101.67269899152149,
  },
  Object {
    "x": 514.4016853440836,
    "y": 149.43737677345445,
  },
  Object {
    "x": 598.575821822178,
    "y": 81.01289688068351,
  },
  Object {
    "x": 593.9691332565161,
    "y": 138.02439185626736,
  },
  Object {
    "x": 646.7667420021913,
    "y": 116.02564985517495,
  },
  Object {
    "x": 607.7024331846321,
    "y": 217.22433797431728,
  },
  Object {
    "x": 648.3180542543278,
    "y": 187.6566220669589,
  },
  Object {
    "x": 688.8940057957145,
    "y": 158.1392023522049,
  },
  Object {
    "x": 724.0517275846853,
    "y": 206.5296549874996,
  },
  Object {
    "x": 683.4403346315265,
    "y": 235.99829378385473,
  },
  Object {
    "x": 751.0860556726084,
    "y": 259.6088671470984,
  },
  Object {
    "x": 713.8486503774006,
    "y": 303.02439185626736,
  },
  Object {
    "x": 666.295824562643,
    "y": 318.4752415750147,
  },
  Object {
    "x": 769.4933492322403,
    "y": 316.2606914995297,
  },
  Object {
    "x": 728.6447581343423,
    "y": 375.1232038059689,
  },
  Object {
    "x": 778.821296488318,
    "y": 375.09305554039594,
  },
]
`;

exports[`parliament chart data overflow 1`] = `
Array [
  Object {
    "x": 16.52287574310691,
    "y": 174.01042968728981,
  },
  Object {
    "x": 59.690263403750805,
    "y": 174.63692508875016,
  },
  Object {
    "x": 26.310647887878105,
    "y": 143.0555147153631,
  },
  Object {
    "x": 103.91652861882618,
    "y": 175.20092303403067,
  },
  Object {
    "x": 72.24502980700855,
    "y": 144.32703776557923,
  },
  Object {
    "x": 41.278078119303956,
    "y": 114.24608437138424,
  },
  Object {
    "x": 99.24608437138426,
    "y": 56.278078119303956,
  },
  Object {
    "x": 129.32703776557923,
    "y": 87.24502980700854,
  },
  Object {
    "x": 128.05551471536313,
    "y": 41.310647887878105,
  },
  Object {
    "x": 160.20092303403067,
    "y": 118.91652861882618,
  },
  Object {
    "x": 159.63692508875025,
    "y": 74.69026340375078,
  },
  Object {
    "x": 159.01042968728984,
    "y": 31.52287574310691,
  },
  Object {
    "x": 240.98957031271019,
    "y": 31.52287574310691,
  },
  Object {
    "x": 240.36307491124978,
    "y": 74.69026340375078,
  },
  Object {
    "x": 271.94448528463687,
    "y": 41.310647887878105,
  },
  Object {
    "x": 239.79907696596933,
    "y": 118.91652861882618,
  },
  Object {
    "x": 270.6729622344208,
    "y": 87.24502980700856,
  },
  Object {
    "x": 300.75391562861574,
    "y": 56.278078119303956,
  },
  Object {
    "x": 358.7219218806961,
    "y": 114.24608437138426,
  },
  Object {
    "x": 327.75497019299144,
    "y": 144.3270377655792,
  },
  Object {
    "x": 373.6893521121219,
    "y": 143.05551471536313,
  },
  Object {
    "x": 296.0834713811738,
    "y": 175.20092303403067,
  },
  Object {
    "x": 340.3097365962492,
    "y": 174.63692508875022,
  },
  Object {
    "x": 383.4771242568931,
    "y": 174.01042968728984,
  },
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
  Object {},
]
`;
